### DISCUSSION TITLE
Surface error messages in dependency dashboard

### DISCUSSION THREAD

**Discussion by morremeyer** [renovatebot/renovate#23807] [Suggest an Idea]
### Tell us more.

We run renovate on-prem (currently version 5.0.0) for all repositories in our GitHub organizations. Not all users have access to our logging platform, so in some cases, help by one of our internal renovate operations people is needed to surface logging information.

Most regularly, these are datasource errors. For example `Failed to look up github-tags package bridgecrewio/checkov`, see log in the details.

<details><summary>Log for datasource error</summary>
<p>

```json
  {
    "name": "renovate",
    "hostname": "renovate-whitesource-renovate-68876df5c4-g8728",
    "pid": 7,
    "level": 20,
    "logContext": "6tmb82Xe_",
    "repository": "redacted/redacted",
    "datasource": "github-tags",
    "packageName": "bridgecrewio/checkov",
    "err": {
      "message": "Although you appear to have the correct authorization credentials, the `bridgecrewio` organization has an IP allow list enabled, and your IP address is not permitted to access this resource.",
      "stack": "Error: Although you appear to have the correct authorization credentials, the `bridgecrewio` organization has an IP allow list enabled, and your IP address is not permitted to access this resource.\n    at GithubGraphqlDatasourceFetcher.doRawQuery (/usr/src/app/node_modules/renovate/dist/util/github/graphql/datasource-fetcher.js:84:29)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async GithubGraphqlDatasourceFetcher.doShrinkableQuery (/usr/src/app/node_modules/renovate/dist/util/github/graphql/datasource-fetcher.js:135:26)\n    at async GithubGraphqlDatasourceFetcher.doPaginatedQuery (/usr/src/app/node_modules/renovate/dist/util/github/graphql/datasource-fetcher.js:167:33)\n    at async GithubGraphqlDatasourceFetcher.getItems (/usr/src/app/node_modules/renovate/dist/util/github/graphql/datasource-fetcher.js:206:21)\n    at async GithubGraphqlDatasourceFetcher.query (/usr/src/app/node_modules/renovate/dist/util/github/graphql/datasource-fetcher.js:31:23)\n    at async queryTags (/usr/src/app/node_modules/renovate/dist/util/github/graphql/index.js:8:17)\n    at async GithubTagsDatasource.getReleases (/usr/src/app/node_modules/renovate/dist/modules/datasource/github-tags/index.js:49:28)\n    at async getRegistryReleases (/usr/src/app/node_modules/renovate/dist/modules/datasource/index.js:61:17)\n    at async huntRegistries (/usr/src/app/node_modules/renovate/dist/modules/datasource/index.js:89:19)\n    at async fetchReleases (/usr/src/app/node_modules/renovate/dist/modules/datasource/index.js:253:23)\n    at async getPkgReleases (/usr/src/app/node_modules/renovate/dist/modules/datasource/index.js:304:34)\n    at async lookupUpdates (/usr/src/app/node_modules/renovate/dist/workers/repository/process/lookup/index.js:51:35)\n    at async withLookupStats (/usr/src/app/node_modules/renovate/dist/workers/repository/process/fetch.js:19:20)\n    at async fetchDepUpdates (/usr/src/app/node_modules/renovate/dist/workers/repository/process/fetch.js:62:38)\n    at async /usr/src/app/node_modules/p-map/index.js:57:22"
    },
    "msg": "Datasource unknown error",
    "time": "2023-08-10T08:49:51.889Z",
    "v": 0
  },
```

</p>
</details> 

However, the Dependency Dashboard only surfaces `Failed to look up github-tags package bridgecrewio/checkov`.

For our users, it would be helpful if it would also surface the `err.message` property. 
I'd love some feedback for this idea, since I might want to implement it myself if it's likely to get merged.

**Comment by rarkins** [comment]
Aren't failed lookups listed in the dashboard anyway?

**Reply by morremeyer** [reply]
Yes, but only with the "Failed to look up" message. For the repository in which the error above occurs, that looks like the screenshot attached.
Ignore the `pypi package python` warning, that's something completely different.

<img width="1123" alt="image" src="https://github.com/renovatebot/renovate/assets/7683567/dba69e7a-f430-4726-9c66-09ea8a85f988">

**Reply by rarkins** [reply]
How much more details were you after?

**Reply by morremeyer** [reply]
The `err.message` property.

**Reply by rarkins** [reply]
I'm not sure there's a clean 1:1 mapping of lookup error to message like that. Many times it's an unhelpful generic http error

**Reply by morremeyer** [reply]
Get it. In that case, I think they would still be helpful to see that it might be a transient error in many cases. At least in our case, most people would be able to determine a timeout error as something that might just need a re-run vs. a "Permission denied" as something that needs action.

We could make this a configurable & experimental behavior and put the messages in a **details** block so that we can give it a try, maybe?

**Reply by rarkins** [reply]
I think the first step is to enhance the lookup results interface so that it can include an optional errorDetails string with any null result. That could be printed as part of the packageFiles with updates log as well as reused for dashboard later

**Reply by morremeyer** [reply]
Quick note of my understanding here:

* Add an optional `errorDetails` string to the [ReleaseResult](https://github.com/renovatebot/renovate/blob/aced1aab16192bda706b065269d5dcb4300ec09e/lib/modules/datasource/types.ts#L66) interface 
* These will be printed in [lookup](https://github.com/renovatebot/renovate/blob/aced1aab16192bda706b065269d5dcb4300ec09e/lib/workers/repository/process/extract-update.ts#L201) since that prints the updates

Is that correct?

**Reply by rarkins** [reply]
Yes, that's what I was thinking

**Reply by morremeyer** [reply]
Coming back to this after a vacation. I'm going to open a PR for this when I get to it. To be clear in my understanding, we'll be adding that field and then fill the `errorDetails` string for the different datasources at places in the code where we know we have a meaningful error message.

Do we share that understanding?
